### PR TITLE
Allow adding draft works to collections

### LINKED ISSUE #1231: Ability to add draft works to a collection
# Feature Request

Users should be able to add drafts work to collections through the collection "works" form. Currently, only published works can be added.

- [x] Draft works that a user has edit access to should appear in the "Add works to collection" search box.
- [x] Adding a draft work causes the work to appear in the list of the works on the "works" form. The work should be labeled as a draft.
- [x] Hint text on the page should read: "Draft works will not appear on the collection's public page until they are published."

Note this issue *does not* require the draft works to be displayed on the collection's public display page.

### PR DESCRIPTION
**Pull request by banukutlu**
Closes #1231

### CODE REVIEW

File: app/views/dashboard/form/members/_search.html.erb [hunk 1]
```diff
@@ -3,7 +3,6 @@
     <div class="form-wrapper form-wrapper--wide">
       <div class="keyline keyline--left mb-3">
         <h4><%= t('dashboard.form.members.search_heading') %></h4>
-      </div>
```
Comment (nicktiberi) [inline app/views/dashboard/form/members/_search.html.erb]: Typo maybe? Looks like now there is a missing closing `</div>` tag.

**Review by nicktiberi — COMMENTED** [review]
This looks great!

I did some local testing and noticed that if you have a collection with 1 draft work in it, and then publish the draft, the collection does not get reindexed, so the collection still won't appear in the main search results. I wonder if we need to just add an action to reindex any collections that the work is a part of as part of publishing a work version?

File: app/views/dashboard/form/members/_search.html.erb (hunk 1 above)
Comment (banukutlu) [reply app/views/dashboard/form/members/_search.html.erb]: yes typo, mistakenly pushed :) thanks for catching it.

Comment (banukutlu) [reply app/views/dashboard/form/members/_search.html.erb]: this is fixed now.

**Comment by banukutlu** [comment]
> This looks great!
> 
> I did some local testing and noticed that if you have a collection with 1 draft work in it, and then publish the draft, the collection does not get reindexed, so the collection still won't appear in the main search results. I wonder if we need to just add an action to reindex any collections that the work is a part of as part of publishing a work version?

that makes sense, I will look into it. thanks for testing this.

**Comment by banukutlu** [comment]
@nicktiberi this is ready for review again :)

**Comment by nicktiberi** [comment]
Awesome! I think this is good to go. Thanks, @banukutlu! 🙂  🚀

**Review by nicktiberi — APPROVED** [review]

**Pull request merged**